subprojects/base-services-groovy/src/main/java/org/gradle/api/internal/ClosureBackedAction.java

```diff
@@ -62,13 +62,19 @@ public void execute(T delegate) {
             if (configureableAware && delegate instanceof Configurable) {
                 ((Configurable) delegate).configure(closure);
             } else {
-                Closure copy = (Closure) closure.clone();
-                copy.setResolveStrategy(resolveStrategy);
-                copy.setDelegate(delegate);
-                if (copy.getMaximumNumberOfParameters() == 0) {
-                    copy.call();
-                } else {
-                    copy.call(delegate);
+                int previousResolveStrategy = closure.getResolveStrategy();
```

Comment:
I removed this change, as it was a problem in any case - in order for our `subprojects {}` DSL to work with nested Closures, each of the nested ones needs its own copy of the parent.